modules/configuration-api/src/main/java/org/apache/ignite/configuration/annotation/ConfigurationExtension.java

```diff
@@ -39,5 +40,12 @@
 @Target(TYPE)
 @Retention(RUNTIME)
 @Documented
-public @interface InternalConfiguration {
+public @interface ConfigurationExtension {
+
+    /**
+     * Controls whether this configuration is part of the public configuration or is hidden from the end user.
+     * The extension is internal by default.
+     */
+    boolean internal() default true;
```

Comment:
Do you think `external` word is a better naming? By default, `external = false` which is common default boolean value.